src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -99,6 +99,27 @@ public void Baggage()
             }
         }
 
+        [Fact]
+        public void TestBaggageOrderAndDuplicateKeys()
```

Comment:
We should probably add a test that confirms baggage enumeration ordering is consistent (across versions) when multiple Activities in a chain have all had baggage added. Looking at the code it appears we have always maintained the order that child activity baggage is enumerated before parent activity baggage.